src/uct/ib/ud/base/ud_inl.h

```diff
@@ -120,32 +120,30 @@ uct_ud_iface_complete_tx_inl(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
                              uct_ud_send_skb_t *skb, void *data,
                              const void *buffer, unsigned length)
 {
+    ucs_time_t now = uct_ud_iface_get_time(iface);
     iface->tx.skb = ucs_mpool_get(&iface->tx.mp);
     ep->tx.psn++;
     skb->len += length;
     memcpy(data, buffer, length);
     ucs_queue_push(&ep->tx.window, &skb->queue);
-    ep->tx.slow_tick = iface->async.slow_tick;
-    ucs_wtimer_add(&iface->async.slow_timer, &ep->slow_timer,
-                   uct_ud_iface_get_async_time(iface) -
-                   ucs_twheel_get_time(&iface->async.slow_timer) +
-                   ep->tx.slow_tick);
-    ep->tx.send_time = uct_ud_iface_get_async_time(iface);
+    ep->tx.tick = iface->tx.tick;
+    ucs_wtimer_add(&iface->tx.timer, &ep->timer,
+                   now - ucs_twheel_get_time(&iface->tx.timer) + ep->tx.tick);
+    ep->tx.send_time = now;
 }
 
 static UCS_F_ALWAYS_INLINE void
 uct_ud_iface_complete_tx_skb(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
                              uct_ud_send_skb_t *skb)
 {
+    ucs_time_t now = uct_ud_iface_get_time(iface);
     iface->tx.skb = ucs_mpool_get(&iface->tx.mp);
     ep->tx.psn++;
     ucs_queue_push(&ep->tx.window, &skb->queue);
-    ep->tx.slow_tick = iface->async.slow_tick;
-    ucs_wtimer_add(&iface->async.slow_timer, &ep->slow_timer,
-                   uct_ud_iface_get_async_time(iface) -
-                   ucs_twheel_get_time(&iface->async.slow_timer) +
-                   ep->tx.slow_tick);
-    ep->tx.send_time = uct_ud_iface_get_async_time(iface);
+    ep->tx.tick = iface->tx.tick;
+    ucs_wtimer_add(&iface->tx.timer, &ep->timer,
+                   now - ucs_twheel_get_time(&iface->tx.timer) + ep->tx.tick);
+    ep->tx.send_time = now;
```

Comment:
can you pls remove some of code duplication here (which existed before this PR)?